pkg/controller/gkenetworkparamset/gkenetworkparamset_controller.go

```diff
@@ -158,9 +151,53 @@ func (c *Controller) Run(numWorkers int, stopCh <-chan struct{}, controllerManag
 		},
 	})
 
+	nodeInformer.Informer().AddEventHandler(cache.ResourceEventHandlerFuncs{
+		AddFunc: func(obj interface{}) {
+			node := obj.(*v1.Node)
+			if c.hasNewPodRange(node) {
+				c.queue.Add(networkv1.DefaultPodNetworkName)
+			}
+		},
+		UpdateFunc: func(old, new interface{}) {
+			oldNode := old.(*v1.Node)
+			newNode := new.(*v1.Node)
+			// the node Pod CIDR could change if the node restarts
+			// controller only reacts after the node Pod CIDR is populated and the new CIDR is different from the old one
+			if oldNode.Spec.PodCIDR != "" && newNode.Spec.PodCIDR != "" && newNode.Spec.PodCIDR != oldNode.Spec.PodCIDR {
```

Comment:
when the node starts spec.PodCIDR is empty, so you need to process when the Node acquires the PodCIDR for the first time too, no?

I think this is something as `if newNode.Spec.PodCIDR != ""  && newNode.Spec.PodCIDR != oldNode.Spec.PodCIDR`